Makefile

```diff
@@ -1,30 +1,44 @@
 containers = grafana graphite-statsd logstash monitoring-proxy sensu sensu-api sensu-client sensu-server uchiwa
 
-currenttag := $(shell semvertag latest)
-newtag := $(shell semvertag bump patch)
+.PHONY: build push pull $(containers) clean showinfo
 
-registryUrl ?= registry.service.opg.digital
-oldRegistryUrl ?= registry.service.dsd.io
+tagrepo = no
+ifdef stage
+	stagearg := --stage $(stage)
+endif
+
+ifdef buildArgs
+	no-cache := --no-cache
+endif
+
+currenttag := $(shell semvertag latest $(stagearg))
+newtag := $(shell semvertag bump patch $(stagearg))
+
+
+registryUrl = registry.service.opg.digital
+oldRegistryUrl = registry.service.dsd.io
 
-.PHONY: build push pull $(containers) clean showinfo
 
 build: $(containers)
+ifeq ($(tagrepo),yes)
```

Comment:
The makefile doesn't have the workaround for semvertag needing an initial tag and also does the tag after the build rather than after the push.